src/controllers/companyController.js

```diff
@@ -0,0 +1,86 @@
+import { companies } from "../models/index.js";
+import NotFoundError from "../errors/notFoundError.js";
+import validateCreateCompany from "../usecases/validateCreateCompany.js";
+
+class CompanyController {
+  static async registerCompany(req, res, next) {
+    try {
+      if (validateCreateCompany(req.body)) {
+        const newCompany = await companies.create(req.body);
+        res.status(201).json({ company: newCompany });
+      }
```

Comment:
Tenho alguns pontos sobre esse ajuste:

1. Pela leitura do código da forma que está, dá a entender que só tem retorno no bloco try se o resultado do método `validateCreateCompany` for true. E creio eu que esse método `validateCreateCompany` só retorna true se a validação for correta, mas não retorna false se a validação falhar, e sim lança erros. 

     Considerando que o método `validateCreateCompany` lança erros caso alguma validação falhe, ele não precisa retornar true caso a validação seja feita com sucesso, pois se a validação falhar, automaticamente o erro lançado vai cair no catch e, consequentemente, ser lançado no `next(err)`. Por isso, esse código pode ficar assim:
     
     ```suggestion
      validateCreateCompany(req.body)

      const newCompany = await companies.create(req.body);
      res.status(201).json({ company: newCompany });
     ```
2. Lembra que o contrato de retorno para uma company criada é este:
    ```javascript
    {
       "id": "" // id, em string, da loja criada
       "name": "", // nome da loja
       "document": "", // cpf ou cnpj da loja
       "city": 10, // id da cidade da loja
       "image": "" // url da imagem da loja
    }
    ```
    Da forma como o retorno está definido, é provável que o retorno desse endpoint esteja assim:
    
    ```javascript
    {
        "company": {
          "id": "" // id, em string, da loja criada
          "name": "", // nome da loja
          "document": "", // cpf ou cnpj da loja
          "city": 10, // id da cidade da loja
          "image": "" // url da imagem da loja
       }
    }    
    ```
    Ou seja, todos os campos da companhia criada devem estar na raiz do objeto, e não dentro de um campo `company`.

Reply:
Bem notado querido!